smarty/environment.py

```diff
@@ -370,32 +366,64 @@ def _getAtomInfo(self, atom):
 
         split = atom.split(';')
 
+        # Get ANDtypes
         ANDtypes = split[1:]
-        ORtypes = split[0].split(',')
-
         if len(ANDtypes) == 0:
             ANDtypes = None
 
-        if len(ORtypes) == 1 and ORtypes[0] == '*':
-            ORtypes = None
-        
+        # Get ORtypes
+        ORList = split[0].split(',')
+        ORtypes = list()
+        # Separate ORtypes into bases and decorators
+        for OR in ORList:
+            ORbase, ORdecors = self._separateORtypes(OR)
+            if ORbase != None:
+                ORtypes.append( (ORbase, ORdecors) )
+
         return ORtypes, ANDtypes, index
 
+    def _separateORtypes(self, ORtype):
+        """
+        Separates ORtype (i.e. "#6X4R+0") into
+        a base and decorators (i.e. '#6', ['X4','R','+0'] )
+        """
+        # special case 1: wild card
+        if ORtype == '*':
+            return None, []
+
+        # Save regular expression for SMIRKS decorators
+        reg = r'(!?[#]\d+|!?[aA]|!?[DHjrVX^]\d+|!?[R+-]\d*|!?[@]\d+|!?[@]@?)'
```

Comment:
I separate the reg into each decorator option with comments and then combined for the actual find all re function.